test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -515,5 +547,156 @@ public void AnomalyDetectionWithSrCnn()
                 k += 1;
             }
         }
+
+        [Fact]
+        public void RootCauseLocalization()
+        {
+            // Create an root cause localizatiom input list.
+            var rootCauseLocalizationData = new List<RootCauseLocalizationData>() { new RootCauseLocalizationData(new DateTime(), new Dictionary<string, Object>(), new List<MetricSlice>() { new MetricSlice(new DateTime(), new List<Microsoft.ML.TimeSeries.Point>()) }, AggregateType.Sum, _aggSymbol) };
+
+            var ml = new MLContext(1);
+            // Convert the list of root cause data to an IDataView object, which is consumable by ML.NET API.
+            var data = ml.Data.LoadFromEnumerable(rootCauseLocalizationData);
+
```

Comment:
Can you please add a test/sample that demonstrates how this would work when loading data from a file?

Reply:
Hi Harish, you mean loading data from a json file or something else? I went through the time series tests and data are loaded  from Enumerable, could you help give more guidance or examples?

Reply:
> Can you please add a test/sample that demonstrates how this would work when loading data from a file?

Hi Harish, you mean loading data from a json file or something else? I went through the time series tests and data are loaded from Enumerable, could you help give more guidance or examples?

Reply:
Hi @suxi-ms , you're right in that there are no tests for time series were data is loaded from a text file. So I've created this PR to add one time series test were data is loaded from text, as well as a couple of other tests where DateTime objects are loaded from text.

https://github.com/dotnet/machinelearning/pull/5063

As you can see it's quiet simple. Typically you'll only need `LoadFromTextFile`, but when working with DateTime objects in particular, you might also need to use a `CustomMappingTransformer` (as shown on my PR).

For LoadFromTextFile you can refer to these links: [[1]](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.textloadersavercatalog.loadfromtextfile?view=ml-dotnet) [[2]](https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/load-data-ml-net) or to the other tests that aren't for time series (several tests use it). For CustomMappingTransformer please refer to [[3]](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.custommappingcatalog.custommapping?view=ml-dotnet).

Reply:
By the way, I believe that LoadFromTextFile will only be able to load and return [these types] (https://github.com/dotnet/machinelearning/blob/bfd2773dd59ac302bf659ed6959738aa68dd5756/src/Microsoft.ML.Core/Data/DataKind.cs#L36-L72)

So in order to load a `RootCauseLocalizationInput` type required by your transformer, I think you'll actually need to use a CustomMappingTransformer similarly to how it is used in my PR above.